src/main/java/org/folio/services/financedata/FinanceDataService.java

```diff
@@ -125,8 +125,8 @@ private void validateFinanceDataCollection(FyFinanceDataCollection financeDataCo
 
   private void calculateAfterAllocation(FyFinanceDataCollection financeDataCollection) {
     financeDataCollection.getFyFinanceData().forEach(financeData -> {
-      var allocationChange = BigDecimal.valueOf(financeData.getBudgetAllocationChange());
-      var currentAllocation = BigDecimal.valueOf(financeData.getBudgetCurrentAllocation());
+      var allocationChange = BigDecimal.valueOf(requireNonNullElse(financeData.getBudgetAllocationChange(), 0.0));
```

Comment:
we need to think about this change. there can be no budget and allocation will be null
![image](https://github.com/user-attachments/assets/56a3699e-3bf8-410d-a8bc-8878b62b84c1)

Reply:
I added a check.